purchase_work_acceptance/models/purchase.py

```diff
@@ -0,0 +1,121 @@
+        comodel_name='work.acceptance.line',
+        inverse_name='purchase_line_id',
+        string="WA Lines",
+        readonly=True,
+        copy=False
+    )
+
+    @api.depends('order_line.wa_lines.wa_id')
+    def _compute_wa(self):
+        for order in self:
+            wa = self.env['work.acceptance']
+            for line in order.order_line:
+                wa |= line.wa_lines.mapped('wa_id')
+            order.wa_ids = wa
+            order.wa_count = len(wa)
+
+    @api.multi
+    def action_view_wa(self):
+        act = self.env.ref('purchase_work_acceptance.action_work_acceptance')
+        result = act.read()[0]
+        create_wa = self.env.context.get('create_wa', False)
+        result['context'] = {
+            'default_purchase_id': self.id,
+            'default_partner_id': self.partner_id.id,
+            'default_partner_ref': self.partner_ref,
+            'default_currency_id': self.currency_id.id,
+            'default_company_id': self.company_id.id,
+            'company_id': self.company_id.id,
+            'default_date_due': self.date_planned,
+            'default_wa_line_ids': [(0, 0, {
+                'purchase_line_id': line.id,
+                'name': line.name,
+                'product_uom': line.product_uom.id,
+                'product_id': line.product_id.id,
+                'price_unit': line.price_unit,
+                'product_qty': line.product_qty - line.qty_accepted,
+                'price_subtotal': line.price_subtotal,
+                }) for line in self.order_line],
+        }
+        if len(self.wa_ids) > 1 and not create_wa:
+            result['domain'] = "[('id', 'in', " + str(self.wa_ids.ids) + ")]"
+        else:
+            res = self.env.ref(
+                'purchase_work_acceptance.view_work_acceptance_form', False)
+            result['views'] = [(res and res.id or False, 'form')]
+            if not create_wa:
+                result['res_id'] = self.wa_ids.id or False
+        result['context']['default_origin'] = self.name
+        result['context']['default_reference'] = self.partner_ref
+        return result
+
+    @api.multi
+    def action_view_invoice(self):
+        create_bill = self.env.context.get('create_bill', False)
+        if create_bill:
+            wizard = self.env.ref(
+                'purchase_work_acceptance.view_select_work_acceptance_wizard')
+            return {
+                'name': _('Work Acceptance'),
+                'type': 'ir.actions.act_window',
+                'view_type': 'form',
+                'view_mode': 'form',
+                'res_model': 'select.work.acceptance.wizard',
+                'views': [(wizard.id, 'form')],
+                'view_id': wizard.id,
+                'target': 'new',
+            }
+        return super(PurchaseOrder, self).action_view_invoice()
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    qty_accepted = fields.Float(
```

Comment:
After some though, I think we can consider removing this field. Please discuss if you see otherwise.
My reasons,
* WA is used only to control the exact quantity in invoice and picking.
* WA can be created and accepted, but never used by invoice or picking. (we intend to make it flexible).